### DISCUSSION TITLE
Various questions on PSM and mass accuracies

### DISCUSSION THREAD

**Discussion by wiraki** [vdemichev/DiaNN#657] [Q&A]
Hello! I am fairly new to proteomics data analysis, so please bear with me.

1) How can I find the number of PSMs identified? I can see the number of precursors and proteins (passing some filters in the stats tsv, but I don't know how to find the number of PSMs.

2) Using MBR and letting DiaNN estimate the mass accuracies, I have noticed that those are estimated twice: during the first pass, and then again during the second pass. And they don't exactly match. I understand that it is preferable to use those average estimates to re-run DiaNN with fixed mass accuracies, but should I then be using the estimate after first, or second pass? 

Furthermore, is it normal to see an estimated MS1 accuracy lower than MS2? E.g. "Mass accuracy = 19ppm, MS1 accuracy = 7ppm"

Thank you in advance!

**Comment by vdemichev** [comment]
1. Easiest is to count the number fo rows in the pr_matrix file or the number of unique 'Precursor.Id' entries in the approriatey filtered main report. PSMs = Precursors in DIA.
2. Any, the difference between these is primarily due to some inherent randomness.

Yes, can happen. Also, these are max tolerances DIA-NN uses, it usually sets them a lot higher than the actual average mass errors.

**Comment by wiraki** [comment]
Hi Vadim,

Thanks for the great software and this quick reply!

>  PSMs = Precursors in DIA

Then don't I already have the answer with the number of Precursors in the Stats report? Also, then the question of how many identified peptides and how many identified PSMs is the same?

> 1. Easiest is to count the number fo rows in the pr_matrix file or the number of unique 'Precursor.Id' entries in the approriatey filtered main report.

~~So, according to the log and the number of rows in the Main report, there are 33.064 IDs. According to pr_matrix, I find 32.585 rows. Then in the Stats report, it says 24.505 precursors for that sample.~~

Edit: I was wrong. Precursor numbers in log, main report and stats report all match to 33.064 IDs. I still can't match that to the 32.585 rows in the `pr_matrix` where there is a value in that sample's column. Can you explain that difference?

**Reply by vdemichev** [reply]
.stats report is filtered in a particular way and presents info per run. If that's what you are looking for, then yes, can use it. If not, the main report provides full flexibility, i.e. can add protein q-value filtering, other types of filtering, etc. pr_matrix has extra filtering, please see the docs.
